units/patch_intent.go

```diff
@@ -469,78 +465,40 @@ func (j *patchIntentProcessor) buildCliPatchDoc(ctx context.Context, patchDoc *p
 	return nil
 }
 
-func (j *patchIntentProcessor) buildBackportPatchDoc(patchDoc *patch.Patch) error {
-	existingMergePatch, err := patch.FindOneId(patchDoc.BackportOf)
-	if err != nil {
-		return errors.Wrap(err, "can't get existing merge patch")
-	}
-	if existingMergePatch == nil {
-		return errors.Errorf("patch '%s' does not exist", patchDoc.BackportOf)
-	}
-	if !existingMergePatch.IsCommitQueuePatch() {
-		return errors.Errorf("can only backport commit queue patches")
-	}
-
-	patchDoc.Patches = existingMergePatch.Patches
-	return nil
-}
-
-// if commit queue then mbox format, organize summaries by commit
-func GetPatchSummariesByCommit(reader io.Reader) ([]patch.Summary, []string, error) {
-	stream, err := mbox.CreateMboxStream(reader)
-	if err != nil {
-		if err == io.EOF {
-			return nil, nil, errors.Errorf("patch is empty")
-		}
-		return nil, nil, errors.Wrap(err, "error creating stream")
-	}
-	if stream == nil {
-		return nil, nil, errors.New("mbox stream is nil")
-	}
-	summaries := []patch.Summary{}
-	commitMessages := []string{}
-	// iterate through patches
-	for err == nil {
-		var buffer []byte
-		msg, err := stream.ReadMessage()
+func (j *patchIntentProcessor) buildBackportPatchDoc(ctx context.Context, projectRef *model.ProjectRef, patchDoc *patch.Patch) error {
+	if len(patchDoc.BackportOf.PatchID) > 0 {
+		existingMergePatch, err := patch.FindOneId(patchDoc.BackportOf.PatchID)
 		if err != nil {
-			if err == io.EOF { // no more patches
-				return summaries, commitMessages, nil
-			}
-			return nil, nil, errors.Wrap(err, "error reading message")
+			return errors.Wrap(err, "can't get existing merge patch")
 		}
-
-		var description string
-		if subject := msg.Headers()["Subject"]; len(subject) > 0 {
-			description = strings.TrimSpace(subject[0])
+		if existingMergePatch == nil {
+			return errors.Errorf("patch '%s' does not exist", patchDoc.BackportOf)
+		}
+		if !existingMergePatch.IsCommitQueuePatch() {
+			return errors.Errorf("can only backport commit queue patches")
 		}
 
-		reader := msg.BodyReader()
-		// iterate through patch body
-		for {
-			curBytes := make([]byte, bytes.MinRead)
-			n, err := reader.Read(curBytes)
-			if err != nil {
-				if err == io.EOF { // finished reading body of this patch
-					var patchSummaries []patch.Summary
-					patchSummaries, err = thirdparty.GetPatchSummaries(string(buffer))
-					if err != nil {
-						return nil, nil, errors.Wrapf(err, "error getting patch summaries for commit '%s'", description)
-					}
-					for i := range patchSummaries {
-						patchSummaries[i].Description = description
-					}
-					summaries = append(summaries, patchSummaries...)
-					commitMessages = append(commitMessages, description)
-					break
-				}
-				return nil, nil, errors.Wrap(err, "error reading body")
+		for _, p := range existingMergePatch.Patches {
+			if p.ModuleName == "" {
+				p.Githash = patchDoc.Githash
```

Comment:
Why do we do this? Also do we need to account for any weirdness with assigning to a variable declared in a loop or is this confirmed to not be weird? (Like potentially doing existingMergePatch.Patches[i].Githash instead)

Reply:
git.get_project [resets to the patch's githash](https://github.com/evergreen-ci/evergreen/blob/89f2312f6dcbb078f7007c58d0da8806304cc82e/command/git.go#L641).

I think it should work given we are storing p's _value_.